### ISSUE TITLE
One-To-Many Association without using a class for associated collection

### ISSUE THREAD

**Issue by nzandy** [linq2db/linq2db#1443]
Apologies in advance if I could have found the answer for this elsewhere - if so, could you point me on how to answer these kinds of questions in future?

I am using explicit mapping using the MappingSchema object - rather than using attribute tags on the class themself. For example I have:

```cs
mappingSchema.GetFluentMappingBuilder()
	.Entity<WaitingList>()
	.HasTableName("WaitingListName")
	.HasSchemaName("Provider")
	.HasPrimaryKey(wl => wl.WaitingListNameId)
	.HasIdentity(wl => wl.WaitingListNameId)
	.Property(wl => wl.Configuration)
	.HasAttribute(new AssociationAttribute { ThisKey = "WaitingListNameId", OtherKey = "WaitingListNameId" });
```

Which is mapping a foreign key (1 to 1) to the "Configuration" property on my WaitingList object. I have the Configuration property also configured as such:

```cs
mappingSchema.GetFluentMappingBuilder()
	.Entity<WaitingListConfiguration>()
	.HasTableName("WaitingList")
	.HasSchemaName("Configuration")
	.HasPrimaryKey(x => x.WaitingListNameId);
```

This works great, I can call using "LoadWith" like the following:
```cs
using (var db = new WaitingListDataConnection())
{
	return db.WaitingList.LoadWith(wl => wl.Configuration)
		.FirstOrDefault(wl => wl.WaitingListNameId == id);
}
```
This works great!

However, I have a mapping table in the database that maps a "WaitingList" to 0 or more "SessionCode" objects. The table looks like this:

```sql
CREATE TABLE [Configuration].[WaitingListNameSessionCode] (
	[WaitingListNameId] INT              NOT NULL,
	[SessionCodeId]     UNIQUEIDENTIFIER NOT NULL,
```

Ideally, what I want to do is add a property to my WaitingList object like:
```cs
List<Guid> LinkedSessionCodes
```
However I can't figure out how I would map this list of Guids...

Basically the SQL query would be something like 
```sql
SELECT SessionCodeId FROM Configuration.WaitingListNameSessionCode
```
The only way I could think of achieving this would be to create a class "WaitingListSessionCode" with a WaitingListNameId and SessionCodeId property.... and then have my WaitingList object contain a property: List<WaitingListSessionCode> and then set up the mapping like I have done for the Configuration property.

My question is - is there any way for me to avoid creating this extra mapping object to store as a property, i.e. can I just declare a property of List<Guid> and somehow map this property to the Correct table?

Thanks in advance!

**Comment by MaceWindu** [comment]
You still need to have mapping class for WaitingListNameSessionCode table to tell linq2db about this table and it's fields.
After that you can configure association of desired type (List<Guid>) using expression.

This article should help you with expressions in associations:
https://github.com/linq2db/linq2db/blob/doc-associations/Doc/articles/model/associations.md

**Comment by nzandy** [comment]
Great thanks for confirming @MaceWindu I have created the mapping class in C# and it works.
